internal/blocksync/reactor.go

```diff
@@ -442,6 +449,11 @@ FOR_LOOP:
 				panic(fmt.Errorf("peeked first block without extended commit at height %d - possible node store corruption", first.Height))
 			}
 
+			select {
+			case <-bcR.pool.Quit():
```

Comment:
q: why did you add `select` to this line specifically?

Reply:
Sorry for the delay in responding.  This `select` is here to prevent a contest between the `Quit` and `didProcessCh` receives and on the outer select in the next iteration.  Just below this select it primes `didProcessCh`, which we want to avoid if we know we're shutting down.  I've just pushed a commit with a comment.

The quirk of `select` `case` with channels is essentially this the "uniform pseudo-random selection" that Go uses when there are multiple cases ready to go: https://play.golang.com/p/GNdVVAdSKIz

Reply:
Should we check for `case <-bcR.Quit():` as well then?

Reply:
That is a good question.  Based on my review of this code (first time in cometbft code however) I don't think any of the `case <-bcR.Quit():` in the other `select`s in this file are even necessary since the `BlockPool`'s quit channel seems to be closed before the `Reactor`'s quit channel.

I think for consistency I should add that `case` however.